srcpkgs/mercury/template

```diff
@@ -0,0 +1,151 @@
+if [ "$XBPS_LIBC" != musl ] ; then
+	_LIBGRADES="${_LIBGRADES},hlc.par.gc,${_LLDS_GRADE}.par.gc.stseg"
+fi
+
+do_configure() {
+	# Prefer the low-level C grade as default, unless it is none.
+	if [ "${_LLDS_GRADE}" != "none" ] ; then
+		DEFAULT_GRADES="${_LLDS_GRADE}.gc.stseg"
+	else
+		DEFAULT_GRADES=hlc.gc
+	fi
+
+	./configure --prefix=/usr/ \
+		--enable-new-mercuryfile-struct \
+		--enable-gc-mmap --enable-gc-munmap \
+		--without-readline --with-editline \
+		--enable-libgrades="${_LIBGRADES}" --with-llds-base-grade="${_LLDS_GRADE}" --with-default-grade="${DEFAULT_GRADES}"
+}
+
+do_build() {
+	# The Mercury makefiles do most of the actual compilation in the 'install'
+	# rule. The 'DESTDIR' makefile variable is used to have the makefile
+	# install to the build directory instead, and then copy the files in
+	# do_install().
+	make PARALLEL="${makejobs}" DESTDIR=$(pwd)
+	make PARALLEL="${makejobs}" DESTDIR=$(pwd) install
+}
+
+do_install() {
+	vmkdir /usr/lib/mercury/lib
+	vmkdir /usr/lib/mercury/conf
+	vman usr/share/man/man1/c2init.1
+	vman usr/share/man/man1/mdb.1
+	vman usr/share/man/man1/mercury_config.1
+	vman usr/share/man/man1/mgnuc.1
+	vman usr/share/man/man1/ml.1
+	vman usr/share/man/man1/mmake.1
+	vman usr/share/man/man1/mmc.1
+	vman usr/share/man/man1/mtags.1
+	vbin usr/bin/c2init
+	vbin usr/bin/canonical_grade
+	vbin usr/bin/mdb
+	vbin usr/bin/mdemangle
+	vbin usr/bin/mercury_compile
+	vbin usr/bin/mercury_config
+	vbin usr/bin/mfiltercc
+	vbin usr/bin/mfilterjavac
+	vbin usr/bin/mgnuc
+	vbin usr/bin/mkinit
+	vbin usr/bin/ml
+	vbin usr/bin/mmake
+	vbin usr/bin/mmc
+	vbin usr/bin/mtags
+	vcopy usr/lib/mercury/conf/Mercury.config /usr/lib/mercury/conf/Mercury.config
+	vcopy usr/lib/mercury/conf/mercury_conf.h /usr/lib/mercury/conf/mercury_conf.h
+	vcopy usr/lib/mercury/ints /usr/lib/mercury/ints
+	vcopy usr/lib/mercury/inc /usr/lib/mercury/inc
+	vcopy usr/lib/mercury/ints /usr/lib/mercury/ints
+	vcopy usr/lib/mercury/mdb /usr/lib/mercury/mdb
+	vcopy usr/lib/mercury/mmake /usr/lib/mercury/mmake
+	vcopy usr/lib/mercury/lib/libgc*.a /usr/lib/mercury/lib
+	for GRADE in $(echo $_LIBGRADES | sed 's/,/ /g') ; do
+		# Skip pregen, we don't need that after bootstrapping.
+		if test "$GRADE" != hlc.gc.pregen ; then
+			vmkdir /usr/lib/mercury/lib/$GRADE
+			vmkdir /usr/lib/mercury/modules/$GRADE
+			vcopy usr/lib/mercury/lib/$GRADE/inc /usr/lib/mercury/lib/$GRADE
+			vcopy usr/lib/mercury/lib/$GRADE/lib*.a /usr/lib/mercury/lib/$GRADE
+			vcopy usr/lib/mercury/modules/$GRADE/*.init usr/lib/mercury/modules/$GRADE
+			# Copy the shared libraries for profiling and debug grades.
+			case $GRADE in
+				*prof*|*debug*)
+					vcopy usr/lib/mercury/lib/$GRADE/lib*.so /usr/lib/mercury/lib/$GRADE
+				;;
+			esac
+		fi
+	done
+}
+
+mercury-libs_package() {
+	short_desc+=" - runtime libraries"
+	noverifyrdeps=yes
+	noshlibprovides=yes
+	pkg_install() {
+		vmkdir /usr/lib/mercury/lib
+		vcopy usr/lib/mercury/lib/libgc*.so /usr/lib/mercury/lib
```

Comment:
this should probably be `vmove`? `vcopy` means that the file will exist in both mercury and mercury-libs, which would make them conflict.

Reply:
The mercury-libs version only copies shared libraries, and the mercury version only copies static libraries, so it should work properly I think? If there was a conflict, I would also see this when trying to install or remove the resulting packages, right?